### DISCUSSION TITLE
Unlock gnome-keyring automatically on login

### DISCUSSION THREAD

**Discussion by heapifyman** [Manjaro-Sway/manjaro-sway#562] [Ideas]
As a follow-up to #530 would it make sense to configure gnome-keyring to unlock the default keyring automatically when logging in?

Use case: some (many?) applications store passwords in gnome-keyring by default. For example, nextcloud sync client, or PCManFM when saving passwords for network shares. Without automatic keyring unlocking one has to enter the keyring password when using an application that gets its passwords from gnome-keyring. If e.g. nextcloud client starts automatically when logging in, that results in:

1. Enter password to login
2. nextcloud client starts and tries to get password from gnome-keyring
3. Enter gnome-keyring password

To skip the 3. step would require adding 2 lines to `/etc/pam.d/greetd` as described [here](https://wiki.archlinux.org/title/GNOME/Keyring#PAM_step) (and 1 line to `etc/pam.d/passwd` as described [here](https://wiki.archlinux.org/title/GNOME/Keyring#Automatically_change_keyring_password_with_user_password)).

As mentioned in the Arch wiki:

> [GDM](https://wiki.archlinux.org/title/GDM), [LightDM](https://wiki.archlinux.org/title/LightDM), [LXDM](https://wiki.archlinux.org/title/LXDM), and [SDDM](https://wiki.archlinux.org/title/SDDM) already have the necessary PAM configuration.

So Manjaro Gnome already automatically unlocks the gnome-keyring, and making this the default also in Manjaro Sway might make it easier for users who are switching to Manjaro Sway.

**Comment by boredland** [comment]
Would you like to create the relevant pam configuration?

**Reply by heapifyman** [reply]
I can try to create a PR

**Reply by boredland** [reply]
for me, this works!
